src/app/project.service.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import { of } from 'rxjs/observable/of';
+import { catchError, map, tap } from 'rxjs/operators';
+
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+
+import { Project } from './models/project';
+
+const httpOptions = {
+  headers: new HttpHeaders({ 'Content-Type': 'application/json' })
+};
+
+@Injectable()
+export class ProjectService {
+  private projectsUrl = 'api/projects'; // URL to web api
+
+  constructor(private http: HttpClient) {}
+
+  getProjects(): Observable<Project[]> {
+    return this.http
+      .get<Project[]>(this.projectsUrl, httpOptions)
+      .pipe(
+        tap(projects => console.log(`fetched projects`)),
+        catchError(this.handleError('getProjects', []))
+      );
+  }
+
+  /**
+   * Handle Http operation that failed.
+   * Let the app continue.
+   * @param operation - name of the operation that failed
+   * @param result - optional value to return as the observable result
+   */
+  private handleError<T>(operation = 'operation', result?: T) {
```

Comment:
considering abstracting this for all http services.

Reply:
Good call.